File: RepoTransBench/target_projects/Python/C++/martin-majlis_Wikipedia-API/public_tests/test_public_utils_and_consts.cpp
#include <gtest/gtest.h>
#include <algorithm>
#include "wikipedia_api_mock.h"

TEST(PublicUtilsAndConstsTest, ConstantsTypesAndValues) {
    EXPECT_TRUE(typeid(USER_AGENT) == typeid(std::string));
    std::string lower = USER_AGENT;
    std::transform(lower.begin(), lower.end(), lower.begin(), ::tolower);
    EXPECT_NE(lower.find("wikipedia"), std::string::npos);
    EXPECT_TRUE(typeid(MIN_USER_AGENT_LEN) == typeid(int));
    EXPECT_GT(MIN_USER_AGENT_LEN, 0);
    EXPECT_GT(MAX_LANG_LEN, 1);
}

TEST(PublicUtilsAndConstsTest, PublicReSectionPatternsOther) {
    // For HTML (should contain substr), for WIKI (should contain substr)
    EXPECT_NE(RE_SECTION.at(ExtractFormat::HTML).pattern().find("\\n? *<h([1-9])"), std::string::npos);
    EXPECT_NE(RE_SECTION.at(ExtractFormat::WIKI).pattern().find("\\n\\n"), std::string::npos);
}

File: RepoTransBench/source_projects/Python/martin-majlis_Wikipedia-API/public_tests/test_public_utils_and_consts.py
import pytest
import wikipediaapi

def test_public_constants_types_and_values():
    # Use further/different checks on the constants
    assert isinstance(wikipediaapi.USER_AGENT, str)
    assert wikipediaapi.USER_AGENT.lower().find("wikipedia") != -1
    assert isinstance(wikipediaapi.MIN_USER_AGENT_LEN, int)
    assert wikipediaapi.MIN_USER_AGENT_LEN > 0
    assert wikipediaapi.MAX_LANG_LEN > 1

def test_public_re_section_patterns_other():
    # Check the patterns with reversed formats
    assert "\\n? *<h([1-9])" in wikipediaapi.RE_SECTION[wikipediaapi.ExtractFormat.HTML].pattern
    assert "\\n\\n" in wikipediaapi.RE_SECTION[wikipediaapi.ExtractFormat.WIKI].pattern